src/workerd/api/crypto/crypto.h

```diff
@@ -666,9 +666,15 @@ class SubtleCrypto: public jsg::Object {
 // DigestStream is a non-standard extension that provides a way of generating
 // a hash digest from streaming data. It combines Web Crypto concepts into a
 // WritableStream and is compatible with both APIs.
+class DigestContext {
+ public:
+  virtual void write(kj::ArrayPtr<kj::byte> buffer) = 0;
```

Comment:
Virtual destructor is required to prevent undefined behavior. 

(I think we should enable clang-tidy for workerd)

Ref: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/virtual-class-destructor.html

Reply:
This isn't really a problem with pure base classes with no data members but I will add a virtual destructor anyway 😄